flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkAppenderFactory.java

```diff
@@ -160,7 +184,8 @@ public EqualityDeleteWriter<RowData> newEqDeleteWriter(
         eqDeleteRowSchema,
         "Equality delete row schema shouldn't be null when creating equality-delete writer");
 
-    MetricsConfig metricsConfig = MetricsConfig.fromProperties(props);
+    MetricsConfig metricsConfig =
+        table != null ? MetricsConfig.forTable(table) : MetricsConfig.fromProperties(props);
```

Comment:
The data row in equality delete should share the same metrics config as the table, so I change this as well.

Reply:
this actually makes a stronger case that we should pass in a valid `Table` object to the `FlinkAppenderFactory` as I mentioned in the other comment. we should also use `forTable` for position delete although it doesn't need schema or SortOrder as you said.